engine/src/main/java/org/terasology/world/internal/EntityAwareWorldProvider.java

```diff
@@ -460,37 +417,37 @@ public void onActivateBlock(OnActivatedComponent event, EntityRef entity) {
     @ReceiveEvent(components = {BlockComponent.class})
     public void onDeactivateBlock(BeforeDeactivateComponent event, EntityRef entity) {
         BlockComponent block = entity.getComponent(BlockComponent.class);
-        if (blockEntityLookup.get(block.position) == entity) {
-            blockEntityLookup.remove(block.position);
+        if (blockEntityLookup.get(block.getPosition(new Vector3i())) == entity) {
+            blockEntityLookup.remove(block.getPosition(new Vector3i()));
         }
     }
 
     @ReceiveEvent(components = {BlockRegionComponent.class})
     public void onBlockRegionActivated(OnActivatedComponent event, EntityRef entity) {
         BlockRegionComponent regionComp = entity.getComponent(BlockRegionComponent.class);
         blockRegions.put(entity, regionComp.region);
-        for (org.joml.Vector3ic pos : regionComp.region) {
-            blockRegionLookup.put(new org.joml.Vector3i(pos), entity);
+        for (Vector3ic pos : regionComp.region) {
+            blockRegionLookup.put(new Vector3i(pos), entity);
```

Comment:
and again with the copying of the key.

if it is a pattern we want to keep, would it be worth making a Map class with some `Map.putWithKeyCopy()` or some such?

Reply:
I don't think this would be worth it, because we cannot easily augment this on existing `Map` implementations.

Reply:
you could in Kotlin 😉

Short of that, though, these lookup maps are private members of the class, so they could switch to being some implementation without much disruption.

Reply:
welp we're not using kotlin. @keturn you will just have to keep dreaming.